### PR TITLE
[BEAM-10961] Enabled strict dependency over sdks-io

### PR DESCRIPTION
**Pull request by sonam-vend**
Since enabling the strict dependencies on all Java modules is a large and critical change to be merged all at once. Therefore, task [BEAM-10961] is divided into sub-tasks and strict dependencies are enabled to each module separately which is comparatively easy to review and merge.**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by aromanenko-dev** [comment]
@sonam-vend Is there any Jira issue(s) about these "Enabled strict dependency ..." PRs?

**Comment by sonam-vend** [comment]
> @sonam-vend Is there any Jira issue(s) about these "Enabled strict dependency ..." PRs?

Yes, there is one https://issues.apache.org/jira/browse/BEAM-10961

**Comment by sonam-vend** [comment]
Run Java PreCommit

**Comment by aromanenko-dev** [comment]
> Yes, there is one https://issues.apache.org/jira/browse/BEAM-10961
@sonam-vend Great! Could you add a prefix with Jira ID (like "[BEAM-10961]") to PRs and commit messages? It will allow automatically link a PR with a correspondent Jira issue and make it easier to observe the history of changes in the future. 

Ptal on this as well:
https://beam.apache.org/contribute/#make-your-change

File: sdks/java/io/kafka/build.gradle
Comment (sonam-vend) [inline sdks/java/io/kafka/build.gradle:66]: For `:sdks:java:container:generateLicenseReport` to succeed, had to exclude kafka-clients and zookeeper.

**Comment by sonam-vend** [comment]
R: @ibzib , @aaltay can you please review or add reviewers who can review this.

File: sdks/java/io/google-cloud-platform/build.gradle
Comment (sonam-vend) [inline sdks/java/io/google-cloud-platform/build.gradle:23]: If I enableStrictDependencies here,I am getting:
```
* What went wrong:
Execution failed for task ':sdks:java:io:google-cloud-platform:analyzeClassesDependencies'.
> No signature of method: ca.cutterslade.gradle.analyze.AnalyzeDependenciesTask$_action_closure3$_closure8.doCall() is applicable for argument types: (org.gradle.api.internal.artifacts.PreResolvedResolvableArtifact) values: [beam-runners-core-java.jar (project :runners:core-java)]
  Possible solutions: findAll(), findAll(), isCase(java.lang.Object), isCase(java.lang.Object)

```

File: sdks/java/io/amazon-web-services2/build.gradle
Comment (ibzib) [inline sdks/java/io/amazon-web-services2/build.gradle:52]:
```suggestion
```

File: sdks/java/io/amazon-web-services2/build.gradle
Comment (ibzib) [inline sdks/java/io/amazon-web-services2/build.gradle:50]: Why are there two `commons-lang` dependencies?

File: sdks/java/io/azure/build.gradle
Comment (ibzib) [inline sdks/java/io/azure/build.gradle:41]:
```suggestion
```

File: sdks/java/io/cassandra/build.gradle
Comment (ibzib) [inline sdks/java/io/cassandra/build.gradle:41]:
```suggestion
```

**Review by ibzib — COMMENTED** [review]
This PR should be broken up further. Many of these changes are trivial and I can approve them right away, but there are also larger changes (like `google-cloud-platform`) that are more complex and need another reviewer.

File: sdks/java/io/google-cloud-platform/build.gradle
Comment (ibzib) [reply sdks/java/io/google-cloud-platform/build.gradle:23]: I ran into this as well. I was able to avoid it temporarily by downgrading from Gradle 6.8 to 6.7, however we will need to figure out a better solution. The first step is to figure out where the `:runners:core-java` dependency is coming from, since it isn't explicitly declared here.

**Comment by shehzaadn-vd** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by shehzaadn-vd** [comment]
> This PR should be broken up further. Many of these changes are trivial and I can approve them right away, but there are also larger changes (like `google-cloud-platform`) that are more complex and need another reviewer.

@ibzib what do you think of this split of this PR: https://docs.google.com/spreadsheets/d/101vYddOUsan25AMVR8_dBSQFCPCY07itMVBrmFXHfZo/edit#gid=0

Basically, I'm proposing a split into 4 PRs. If you would prefer a different split, please mark the sheet accordingly.

**Comment by shehzaadn-vd** [comment]
> > This PR should be broken up further. Many of these changes are trivial and I can approve them right away, but there are also larger changes (like `google-cloud-platform`) that are more complex and need another reviewer.
> 
> @ibzib what do you think of this split of this PR: https://docs.google.com/spreadsheets/d/101vYddOUsan25AMVR8_dBSQFCPCY07itMVBrmFXHfZo/edit#gid=0
> 
> Basically, I'm proposing a split into 4 PRs. If you would prefer a different split, please mark the sheet accordingly.

@ibzib I've created https://github.com/apache/beam/pull/13791 for the GCP changes.

File: sdks/java/io/kinesis/build.gradle
Comment (shehzaadn-vd) [inline sdks/java/io/kinesis/build.gradle:45]:
```suggestion
```

File: sdks/java/io/clickhouse/build.gradle
Comment (shehzaadn-vd) [inline sdks/java/io/clickhouse/build.gradle:62]:
```suggestion
```

**Review by shehzaadn-vd — CHANGES_REQUESTED** [review]

File: sdks/java/io/splunk/build.gradle
Comment (shehzaadn-vd) [inline sdks/java/io/splunk/build.gradle:38]:
```suggestion
```

**Review by shehzaadn-vd — CHANGES_REQUESTED** [review]

**Comment by ibzib** [comment]
> Basically, I'm proposing a split into 4 PRs. If you would prefer a different split, please mark the sheet accordingly.

@shehzaadn-vd How about this:

- There are a number of files where the only change is setting `enableStrictDependencies: true`, or others where the only changes are setting `enableStrictDependencies: true` and removing `hamcrest_core` and `hamcrest_library`. These are fine to bundle in a single PR because they are trivial to review (I would LGTM immediately).
- Any files with larger diffs than that should go in their own PR, with the exception of the following groups, which can get one PR each: amazon, hadoop, and snowflake.

Another thought: it's also fine to set `enableStrictDependencies` on a file and just `permit` every existing violation. From the description in the JIRA, it seems the original intent was to first set `enableStrictDependencies` globally and ignore all the existing violations, and second to gradually fix all the existing violations. Step 1 is way easier than step 2 and has the benefit of preventing future violations. Separating the two steps also has the benefit that the `permit` clauses inserted in step 1 make reviewing step 2 a lot easier. It seems we've skipped to step 2 here. Just something to keep in mind in case step 2 is too difficult for some modules. cc @tysonjh

File: sdks/java/io/bigquery-io-perf-tests/build.gradle
Comment (shehzaadn-vd) [inline sdks/java/io/bigquery-io-perf-tests/build.gradle:28]: No need for compile+permitUnusedDeclared here since this bigquery-io-perf-tests just contains tests.

File: sdks/java/io/contextualtextio/build.gradle
Comment (shehzaadn-vd) [inline sdks/java/io/contextualtextio/build.gradle:28]: "compile" lines have just been moved, not removed.

**Comment by shehzaadn-vd** [comment]
@ibzib do you think this is good to go?

**Review by ibzib — COMMENTED** [review]
I'd still prefer this PR to be split up further, with one file per PR (with the exception of kinesis and kinesis/expansion-service, which are fine to put in the same PR if you like). These files are totally unrelated but for the fact that they are IOs, so it will be much easier to review each file separately. Many of these files are ready to merge right now, like in #13902. But when files that are ready to merged are mixed in with files that are not, it becomes very time-consuming to review, since every time I have to re-evaluate which files are ready and which need changes. Plus, it is a lot easier to roll back smaller changes in case we accidentally create a bug.

**Comment by shehzaadn-vd** [comment]
> I'd still prefer this PR to be split up further, with one file per PR (with the exception of kinesis and kinesis/expansion-service, which are fine to put in the same PR if you like). These files are totally unrelated but for the fact that they are IOs, so it will be much easier to review each file separately. Many of these files are ready to merge right now, like in #13902. But when files that are ready to merged are mixed in with files that are not, it becomes very time-consuming to review, since every time I have to re-evaluate which files are ready and which need changes. Plus, it is a lot easier to roll back smaller changes in case we accidentally create a bug.

@ibzib Point well taken. 21 PRs coming up :-)

**Comment by ibzib** [comment]
This PR was broken up and merged in pieces.

**Pull request closed without merging**
